app/src/main/java/org/mozilla/fenix/onboarding/view/Onboarding.kt

```diff
@@ -83,7 +85,16 @@ fun Onboarding(
             modifier = Modifier.fillMaxWidth(),
             horizontalArrangement = Arrangement.End,
         ) {
-            IconButton(onClick = onDismiss) {
+            IconButton(
+                onClick = {
+                    if (onboardingState == OnboardingState.Welcome) {
+                        OnboardingMetrics.welcomeCloseClicked.record(NoExtras())
```

Comment:
I'm shocked that the Glean metrics don't break the Compose previews lol. I feel the metrics could live within `HomeOnboardingDialogFragment` though. @gabrielluong, do you have a strong opinion on this?

Reply:
I think this is fine for now. I would want to look at refactoring some of this and make sure of Interactor so that we can test this, but that can be a followup after this lands.